src/_pytest/_io/saferepr.py

```diff
@@ -1,8 +1,29 @@
-import sys
+import pprint
 
 from six.moves import reprlib
 
 
+def _call_and_format_exception(call, x, *args):
+    try:
+        # Try the vanilla repr and make sure that the result is a string
+        return call(x, *args)
+    except Exception as exc:
+        try:
+            exc_name = exc.__class__.__name__
+        except Exception:
+            exc_name = "unknown"
```

Comment:
This is not covered in tests (https://codecov.io/gh/pytest-dev/pytest/compare/9eac4733c56cd1bac8abb6639e19976710fee139...4024a121e446f348c5ea839e002f3362b8c4978a/diff#D1-13).
And I think it is not possible to get an exception via this, is it?  (/cc @asottile)

Reply:
sadly it is:

```python
class CryParty(object):
    @property
    def __class__(self):
        raise TypeError('boom!')
```

```pycon
>>> CryParty()
<__main__.CryParty object at 0x7f2e3c41b8d0>
>>> CryParty().__class__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "t.py", line 4, in __class__
    raise TypeError('boom!')
TypeError: boom!
>>> type(CryParty()).__name__
'CryParty'
```

Reply:
Awesome! Thank you so much! I saw a decrease in the coverage, and I was looking without any luck for the whole evening to find a way to raise an exception with `__class__`. So I was even going to remove `try .. except` block.